app/uk/gov/hmrc/tai/service/journeyCompletion/JourneyCompletionService.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tai.service.journeyCompletion
+
+import com.google.inject.Inject
+import com.google.inject.name.Named
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.tai.service.journeyCache.JourneyCacheService
+import uk.gov.hmrc.tai.util.constants.JourneyCacheConstants
+import scala.concurrent.ExecutionContext.Implicits.global
+
+import scala.concurrent.Future
+
+abstract class JourneyCompletionService(successfulJourneyCacheService: JourneyCacheService) extends JourneyCacheConstants {
```

Comment:
Does this need to be an extract class? It's only extended in one place

Reply:
Was potentially expecting more classes to come along an extend. The duplicate warning logic could also make use of the new functionality in a refactor